### PR TITLE
add darkphotonDimuon cards

### PR DESCRIPTION
**Pull request by y19y19**
Please check the cards under production/13TeV/

### CODE REVIEW

File: bin/MadGraph5_aMCatNLO/cards/production/13TeV/darkphotonDimuon/darkphotonDimuon_DPMass5e-1MeV_DF1e-2/darkphotonDimuon_DPMass5e-1MeV_DF1e-2_run_card.dat [hunk 1]
```diff
@@ -0,0 +1,283 @@
...
+#*********************************************************************
+# Tag name for the run (one word)                                    *
+#*********************************************************************
+  tag_1     = run_tag ! name of the run 
+#*********************************************************************
+# Number of events and rnd seed                                      *
+# Warning: Do not generate more than 1M events in a single run       *
+# If you want to run Pythia, avoid more than 50k events in a run.    *
+#*********************************************************************
+  50000 = nevents ! Number of unweighted events requested
```
Comment (sihyunjeon) [inline bin/MadGraph5_aMCatNLO/cards/production/13TeV/darkphotonDimuon/darkphotonDimuon_DPMass5e-1MeV_DF1e-2/darkphotonDimuon_DPMass5e-1MeV_DF1e-2_run_card.dat:31]: You don't need to run 50K events for pilot run.
2-3K would be enough

File: bin/MadGraph5_aMCatNLO/cards/production/13TeV/darkphotonDimuon/darkphotonDimuon_DPMass5e-1MeV_DF1e-2/darkphotonDimuon_DPMass5e-1MeV_DF1e-2_run_card.dat [hunk 2]
```diff
@@ -0,0 +1,283 @@
...
+#*********************************************************************
+ 5 = maxjetflavor    ! Maximum jet pdg code
+#*********************************************************************
+#
+#*********************************************************************
+# Store info for systematics studies                                 *
+# WARNING: Do not use for interference type of computation           *
+#*********************************************************************
+   True  = use_syst      ! Enable systematics studies
+#
```
Comment (sihyunjeon) [inline bin/MadGraph5_aMCatNLO/cards/production/13TeV/darkphotonDimuon/darkphotonDimuon_DPMass5e-1MeV_DF1e-2/darkphotonDimuon_DPMass5e-1MeV_DF1e-2_run_card.dat:280]: Remove lines below from here, won't be used

File: bin/MadGraph5_aMCatNLO/cards/production/13TeV/darkphotonDimuon/darkphotonDimuon_DPMass5e-1MeV_DF1e-2/darkphotonDimuon_DPMass5e-1MeV_DF1e-2_run_card.dat [hunk 3]
```diff
@@ -0,0 +1,283 @@
...
+ 0.0  = etaamin ! min rap for the photons
+ 0.0  = etalmin ! main rap for the charged leptons
+ {} = eta_min_pdg ! rap cut for other particles (use pdg code). Applied on particle and anti-particle
+ {1023: 2.5} = eta_max_pdg ! rap cut for other particles (syntax e.g. {6: 2.5, 23: 5})
+#*********************************************************************
+# Minimum and maximum DeltaR distance                                *
+#*********************************************************************
+ 0.4 = drjj    ! min distance between jets 
+ 0.0   = drbb    ! min distance between b's 
+ 0.4 = drll    ! min distance between leptons
```
Comment (sihyunjeon) [inline bin/MadGraph5_aMCatNLO/cards/production/13TeV/darkphotonDimuon/darkphotonDimuon_DPMass5e-1MeV_DF1e-2/darkphotonDimuon_DPMass5e-1MeV_DF1e-2_run_card.dat:157]: Can we go a bit more easy on eatl and drll cuts, there would be (though small) some events smeared after reconstruction?

File: bin/MadGraph5_aMCatNLO/cards/production/13TeV/darkphotonDimuon/darkphotonDimuon_DPMass5e-1MeV_DF1e-2/darkphotonDimuon_DPMass5e-1MeV_DF1e-2_run_card.dat [hunk 4]
```diff
@@ -0,0 +1,283 @@
...
+   False  = cut_decays    ! Cut decay products 
+#*********************************************************************
+# Standard Cuts                                                      *
+#*********************************************************************
+# Minimum and maximum pt's (for max, -1 means no cut)                *
+#*********************************************************************
+ 20.0  = ptj       ! minimum pt for the jets 
+ 0.0  = ptb       ! minimum pt for the b 
+ 10.0  = pta       ! minimum pt for the photons 
+ 10.0  = ptl       ! minimum pt for the charged leptons
```
Comment (sihyunjeon) [inline bin/MadGraph5_aMCatNLO/cards/production/13TeV/darkphotonDimuon/darkphotonDimuon_DPMass5e-1MeV_DF1e-2/darkphotonDimuon_DPMass5e-1MeV_DF1e-2_run_card.dat:117]: reco level ptcuts are higher than 10 GeV for both muons?

**Comment by sihyunjeon** [comment]
Can you specify the lines you modified in the cuts.f file?

**Comment by y19y19** [comment]
Okay cool. It is the amount of events that I’ve generated locally. I can change it.

**Comment by y19y19** [comment]
Sorry but I don’t quite get why we are worrying about reconstruction in the generation step? Do you mean that if a lepton leaves the primary vertex at an angle about eta > 2.5, it might end up being detected with eta < 2.5 due to the magnetic field? Feel free to correct me. 

The signal we are looking for is 2 tight muons, and better to be separated because we are using IsoMu triggers. Actually it does not matter much, because the two muons are mostly back to back.

Thanks

**Comment by y19y19** [comment]
Yes, in the signal region selection, we choose both muon that has pt > 10 GeV.

**Comment by y19y19** [comment]
Yes. So I defined some variables from line 150 to line 154, and I define the selection from line 460 to 498. Other lines are default cut.f file.

**Comment by sihyunjeon** [comment]
About the cuts,
There could be some smearings (ME level ptj = 9 GeV could be reconstructed as in RECO level 11 GeV, things like this). If you feel okay with the cuts I won't really push but I usually ask just to check.

**Comment by y19y19** [comment]
We are not generating any jet in our signal, so I guess the ptj in the card does not affect us?

**Comment by sihyunjeon** [comment]
Sorry, I meant ptl. But if you want to keep it tight, it's okay for me.
@agrohsje Can you check these cards?

**Comment by y19y19** [comment]
Sure. And for ptl, it shouldn’t matter much. But thank you for reminding me about this. If I found there is smearing effect in the signal’s muons in reco, I will apply the cut on my backgrounds.

**Comment by agrohsje** [comment]
I still see many unresolved comments. @y19y19 can you please fix?

File: bin/MadGraph5_aMCatNLO/cards/production/13TeV/darkphotonDimuon/darkphotonDimuon_DPMass5e-1MeV_DF1e-2/darkphotonDimuon_DPMass5e-1MeV_DF1e-2_customizecards.dat [hunk 5]
```diff
@@ -0,0 +1,6 @@
+set param_card HIDDEN 1 5e-4
+set param_card HIDDEN 2 1e4
+set param_card HIDDEN 3 1e-1
+set param_card HIDDEN 4 1e-10
+set param_card DECAY 35 auto
```
Comment (agrohsje) [inline bin/MadGraph5_aMCatNLO/cards/production/13TeV/darkphotonDimuon/darkphotonDimuon_DPMass5e-1MeV_DF1e-2/darkphotonDimuon_DPMass5e-1MeV_DF1e-2_customizecards.dat:5]: where do you set the masses of 35 and 1023 ?

File: bin/MadGraph5_aMCatNLO/cards/production/13TeV/darkphotonDimuon/darkphotonDimuon_DPMass5e-1MeV_DF1e-2/darkphotonDimuon_DPMass5e-1MeV_DF1e-2_cuts.f [hunk 6]
```diff
@@ -0,0 +1,1794 @@
...
+C     pT definition!
+      if( .not. allowed) temp = 1d15
+
+      if(temp .lt. 0.0) temp = 0.0
+
+C     Return pT
+      rhoPythia = dsqrt(temp);
+
+      RETURN
+      END
```
Comment (agrohsje) [inline bin/MadGraph5_aMCatNLO/cards/production/13TeV/darkphotonDimuon/darkphotonDimuon_DPMass5e-1MeV_DF1e-2/darkphotonDimuon_DPMass5e-1MeV_DF1e-2_cuts.f:1794]: Did you take the file from 2.6.5?

File: bin/MadGraph5_aMCatNLO/cards/production/13TeV/darkphotonDimuon/darkphotonDimuon_DPMass5e-1MeV_DF1e-2/darkphotonDimuon_DPMass5e-1MeV_DF1e-2_proc_card.dat [hunk 7]
```diff
@@ -0,0 +1,10 @@
+import model --modelname  HAHM_MG5model_v3/HAHM_variableMW_v3_UFO
+
+define p = g u c d s b u~ c~ d~ s~ b~
+define j = g u c d s u~ c~ d~ s~
+define l+ = e+ m+
+define l- = e- m-
```
Comment (agrohsje) [inline bin/MadGraph5_aMCatNLO/cards/production/13TeV/darkphotonDimuon/darkphotonDimuon_DPMass5e-1MeV_DF1e-2/darkphotonDimuon_DPMass5e-1MeV_DF1e-2_proc_card.dat:6]: why is this needed ?

**Comment by y19y19** [comment]
Yes I will fix all by today. Thanks.

**Comment by y19y19** [comment]
Madgraph is not happy with the HAHM model that removes mu+ and mu-, and then replaces them with m+ and m-. It requests me to redefine the l+ and l-, even though I am not using them in my model.

**Comment by y19y19** [comment]
The hidden parameter 1 is the masss of 1023, hidden parameter 2 is the mass of 35.

**Comment by y19y19** [comment]
I will make the changes and make a new pull request to the master branch

**Comment by agrohsje** [comment]
You mean you will update this PR? Not a new PR, right?

**Comment by y19y19** [comment]
Since my pull request is toward mg261, and I think I have to make a new PR to get to push into master branch. If you would like me to updates all the fixes here before moving to master branch, please let me know.

File: bin/MadGraph5_aMCatNLO/cards/production/13TeV/darkphotonDimuon/darkphotonDimuon_DPMass5e-1MeV_DF1e-2/darkphotonDimuon_DPMass5e-1MeV_DF1e-2_run_card.dat (hunk 1 above)
Comment (y19y19) [reply bin/MadGraph5_aMCatNLO/cards/production/13TeV/darkphotonDimuon/darkphotonDimuon_DPMass5e-1MeV_DF1e-2/darkphotonDimuon_DPMass5e-1MeV_DF1e-2_run_card.dat:31]: yes I will fix this

File: bin/MadGraph5_aMCatNLO/cards/production/13TeV/darkphotonDimuon/darkphotonDimuon_DPMass5e-1MeV_DF1e-2/darkphotonDimuon_DPMass5e-1MeV_DF1e-2_run_card.dat (hunk 2 above)
Comment (y19y19) [reply bin/MadGraph5_aMCatNLO/cards/production/13TeV/darkphotonDimuon/darkphotonDimuon_DPMass5e-1MeV_DF1e-2/darkphotonDimuon_DPMass5e-1MeV_DF1e-2_run_card.dat:280]: yes I will delete these lines 277-280.

File: bin/MadGraph5_aMCatNLO/cards/production/13TeV/darkphotonDimuon/darkphotonDimuon_DPMass5e-1MeV_DF1e-2/darkphotonDimuon_DPMass5e-1MeV_DF1e-2_run_card.dat (hunk 4 above)
Comment (y19y19) [reply bin/MadGraph5_aMCatNLO/cards/production/13TeV/darkphotonDimuon/darkphotonDimuon_DPMass5e-1MeV_DF1e-2/darkphotonDimuon_DPMass5e-1MeV_DF1e-2_run_card.dat:117]: Yes and we are not planning to change that. We will be aware of the smearing effect on muon in the reco.

File: bin/MadGraph5_aMCatNLO/cards/production/13TeV/darkphotonDimuon/darkphotonDimuon_DPMass5e-1MeV_DF1e-2/darkphotonDimuon_DPMass5e-1MeV_DF1e-2_customizecards.dat (hunk 5 above)
Comment (y19y19) [reply bin/MadGraph5_aMCatNLO/cards/production/13TeV/darkphotonDimuon/darkphotonDimuon_DPMass5e-1MeV_DF1e-2/darkphotonDimuon_DPMass5e-1MeV_DF1e-2_customizecards.dat:5]: The hidden parameter 1 is the masss of 1023, hidden parameter 2 is the mass of 35.

File: bin/MadGraph5_aMCatNLO/cards/production/13TeV/darkphotonDimuon/darkphotonDimuon_DPMass5e-1MeV_DF1e-2/darkphotonDimuon_DPMass5e-1MeV_DF1e-2_proc_card.dat (hunk 7 above)
Comment (y19y19) [reply bin/MadGraph5_aMCatNLO/cards/production/13TeV/darkphotonDimuon/darkphotonDimuon_DPMass5e-1MeV_DF1e-2/darkphotonDimuon_DPMass5e-1MeV_DF1e-2_proc_card.dat:6]: Madgraph is not happy with the HAHM model that removes mu+ and mu-, and then replaces them with m+ and m-. It requests me to redefine the l+ and l-, even though I am not using them in my model.

File: bin/MadGraph5_aMCatNLO/cards/production/13TeV/darkphotonDimuon/darkphotonDimuon_DPMass5e-1MeV_DF1e-2/darkphotonDimuon_DPMass5e-1MeV_DF1e-2_run_card.dat (hunk 3 above)
Comment (y19y19) [reply bin/MadGraph5_aMCatNLO/cards/production/13TeV/darkphotonDimuon/darkphotonDimuon_DPMass5e-1MeV_DF1e-2/darkphotonDimuon_DPMass5e-1MeV_DF1e-2_run_card.dat:157]: Do you have a recommendation for what usually is done with muons? what usually do you set for drll and eta range for it? I don't mind to loose it a bit.

File: bin/MadGraph5_aMCatNLO/cards/production/13TeV/darkphotonDimuon/darkphotonDimuon_DPMass5e-1MeV_DF1e-2/darkphotonDimuon_DPMass5e-1MeV_DF1e-2_cuts.f (hunk 6 above)
Comment (y19y19) [reply bin/MadGraph5_aMCatNLO/cards/production/13TeV/darkphotonDimuon/darkphotonDimuon_DPMass5e-1MeV_DF1e-2/darkphotonDimuon_DPMass5e-1MeV_DF1e-2_cuts.f:1794]: I think I took the file from 2.6.1, but not 100% sure.

**Comment by sihyunjeon** [comment]
@y19y19 I think it's fine to make a PR to the master branch and close this one if you update the things discussed here.

File: bin/MadGraph5_aMCatNLO/cards/production/13TeV/darkphotonDimuon/darkphotonDimuon_DPMass5e-1MeV_DF1e-2/darkphotonDimuon_DPMass5e-1MeV_DF1e-2_proc_card.dat (hunk 7 above)
Comment (sihyunjeon) [reply bin/MadGraph5_aMCatNLO/cards/production/13TeV/darkphotonDimuon/darkphotonDimuon_DPMass5e-1MeV_DF1e-2/darkphotonDimuon_DPMass5e-1MeV_DF1e-2_proc_card.dat:6]: @y19y19 If you remove --modelname it would be okay. Is there any reason to use this option?

**Comment by y19y19** [comment]
yes I will get the cut.f from a naive gridpack run with my cards using MGv265.

File: bin/MadGraph5_aMCatNLO/cards/production/13TeV/darkphotonDimuon/darkphotonDimuon_DPMass5e-1MeV_DF1e-2/darkphotonDimuon_DPMass5e-1MeV_DF1e-2_cuts.f (hunk 6 above)
Comment (y19y19) [reply bin/MadGraph5_aMCatNLO/cards/production/13TeV/darkphotonDimuon/darkphotonDimuon_DPMass5e-1MeV_DF1e-2/darkphotonDimuon_DPMass5e-1MeV_DF1e-2_cuts.f:1794]: I will take the one from 2.6.5 in the new pull request to master branch

File: bin/MadGraph5_aMCatNLO/cards/production/13TeV/darkphotonDimuon/darkphotonDimuon_DPMass5e-1MeV_DF1e-2/darkphotonDimuon_DPMass5e-1MeV_DF1e-2_run_card.dat (hunk 4 above)
Comment (y19y19) [reply bin/MadGraph5_aMCatNLO/cards/production/13TeV/darkphotonDimuon/darkphotonDimuon_DPMass5e-1MeV_DF1e-2/darkphotonDimuon_DPMass5e-1MeV_DF1e-2_run_card.dat:117]: I will loose it to 8 GeV for minimum pt of charged leptons

File: bin/MadGraph5_aMCatNLO/cards/production/13TeV/darkphotonDimuon/darkphotonDimuon_DPMass5e-1MeV_DF1e-2/darkphotonDimuon_DPMass5e-1MeV_DF1e-2_run_card.dat (hunk 3 above)
Comment (y19y19) [reply bin/MadGraph5_aMCatNLO/cards/production/13TeV/darkphotonDimuon/darkphotonDimuon_DPMass5e-1MeV_DF1e-2/darkphotonDimuon_DPMass5e-1MeV_DF1e-2_run_card.dat:157]: Loosing drll does not make a difference in the cross section, because our signal is three body final state, so muons are unlikely boosted to the same direction. I will loose the eta of muons to 2.8.

File: bin/MadGraph5_aMCatNLO/cards/production/13TeV/darkphotonDimuon/darkphotonDimuon_DPMass5e-1MeV_DF1e-2/darkphotonDimuon_DPMass5e-1MeV_DF1e-2_proc_card.dat (hunk 7 above)
Comment (y19y19) [reply bin/MadGraph5_aMCatNLO/cards/production/13TeV/darkphotonDimuon/darkphotonDimuon_DPMass5e-1MeV_DF1e-2/darkphotonDimuon_DPMass5e-1MeV_DF1e-2_proc_card.dat:6]: if I remove --modelname, madgraph also complains:
The import format was not given, so we guess it as command
import model HAHM_MG5model_v3/HAHM_variableMW_v3_UFO
INFO: failed to load model from pickle file. Try importing UFO from File 
Command "import /uscms_data/d3/yyao/DarkPhoton_2/Gridpack2/genproductions/bin/MadGraph5_aMCatNLO/darkphotonDimuon_DPMass5e-1MeV_DF1e-2/darkphotonDimuon_DPMass5e-1MeV_DF1e-2_gridpack/work/darkphotonDimuon_DPMass5e-1MeV_DF1e-2_proc_card.dat" interrupted in sub-command:
"define higgs = h hs" with error:
InvalidCmd : No particle hs in model

I am not sure how to make it work. Let me know if you have an idea.

**Comment by agrohsje** [comment]
@y19y19 will you make a cleaned-up request to master? should i close this one?

**Comment by y19y19** [comment]
Yes Please close the request to mg261, I made a new pull request to master branch. We can continue discussion there.

Thanks

**Comment by sihyunjeon** [comment]
closing

**Comment by sihyunjeon** [comment]
#2863 Followup from here
